### PR TITLE
Sync bundler and rubygems

### PR DESCRIPTION
**Pull request by deivid-rodriguez**

### CODE REVIEW

**Comment by simi** [comment]
Is https://github.com/ruby/ruby/pull/4532 related?

**Comment by deivid-rodriguez** [comment]
It's a duplicate, oops! I'm just doing this because I need https://github.com/ruby/ruby/pull/4533/commits/932e47298b12b97fdb76702d7119f53f763b3500 in order to get https://github.com/rubygems/rubygems/pull/4628 green.

File: test/rubygems/test_gem_commands_open_command.rb [hunk 1]
```diff
@@ -21,7 +21,7 @@ def gem(name, version = "1.0")
 
   def test_execute
     @cmd.options[:args] = %w[foo]
-    @cmd.options[:editor] = "#{Gem.ruby} -eexit --"
```
Comment (hsbt) [inline test/rubygems/test_gem_commands_open_command.rb:24]: Do not overwrite this https://github.com/ruby/ruby/commit/ceea1ea8af2376845f5f6164e9432e5b9845c033

Comment (deivid-rodriguez) [reply test/rubygems/test_gem_commands_open_command.rb:24]: I have already ported ceea1ea at https://github.com/rubygems/rubygems/commit/d35448a98c86e198860a4b53c16873941d07b790, so I can confirm I'm not planning to revert it :)

**Comment by hsbt** [comment]
@deivid-rodriguez Thanks for your work. I synced https://github.com/rubygems/rubygems/pull/4611 and https://github.com/rubygems/rubygems/pull/4599 to the current master branch of ruby/ruby because they changed many of files.

**Comment by deivid-rodriguez** [comment]
@hsbt There was some issues on MacOS that I have fixed upstream, so this is now ready. I will release a new version with those fixes so that you can backport it to https://github.com/ruby/ruby/pull/4569 as well.

**Comment by deivid-rodriguez** [comment]
Thanks!

**Pull request merged**